rosnav/model/agent_factory.py

```diff
@@ -54,8 +54,13 @@ def instantiate(cls, name: str, **kwargs) -> Union[Type[BaseAgent], Type[BasePol
         """
         assert name in cls.registry, f"Agent '{name}' is not registered!"
         agent_class = cls.registry[name]
+
+        path=kwargs.pop('path', None)
         
         if issubclass(agent_class, BaseAgent):
             return agent_class(**kwargs)
+        elif name == "CUSTOM":
+            setattr(agent_class, "path", path)
+            return agent_class
```

Comment:
DAs brauchst du eigentlich alles gar nicht. Du hast den Path ja schon in den args. Das kannst du dann einfach in der init als argument erwarten